mtg_card_identifier/RootView.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  RootView.swift
+//  mtg_card_identifier
+//
+//  Created by Dylan Martin on 8/11/23.
+//
+
+import SwiftUI
+import ComposableArchitecture
+
+struct RootViewReducer: Reducer {
+    struct State: Equatable {
+        var isDatabaseDownloaded = SQLiteFileManager.checkDatabaseStatus()
+        var downloadDatabaseState = DownloadDatabaseFilesViewReducer.State()
+    }
+    
+    enum Action: BindableAction {
+        case binding(BindingAction<State>)
+        case downloadDatabaseFilesViewAction(DownloadDatabaseFilesViewReducer.Action)
+    }
+    
+    var body: some Reducer<State, Action> {
+        BindingReducer()
+        
+        Scope(state: \.downloadDatabaseState, action: /Action.downloadDatabaseFilesViewAction) {
+          DownloadDatabaseFilesViewReducer()
+        }
+        
+        Reduce { state, action in
+            switch action {
+            case .binding:
+                return .none
+            case .downloadDatabaseFilesViewAction:
+                return .none
+            }
+        }
+    }
+}
+struct RootView: View {
+    let store: StoreOf<RootViewReducer>
+    
+    var body: some View {
+        WithViewStore(store, observe: { $0 }) { viewStore in
```

Comment:
Technically we only need to observe `$0.isDatabaseDownloaded` not the whole `$0`, but currently that doesn't matter too much.